Component/category item

abdo-essam opened the pull request:
Make sure that you add the count and is selected as an optional parameter. as we can use it in multiple ways as needed.
Ensure that your component has an optional parameter for Modifier.
You must add a preview annotation we have made to preview your components.
![image](https://github.com/user-attachments/assets/a56b1260-a773-40c6-89dc-879ebc6da9af)

abdo-essam committed:
feat: Implement CategoryItem component

This commit introduces the `CategoryItem` composable, a reusable UI
component for displaying category information. It includes an icon,
title, and optional count and selection indicator.  It also adds
the necessary vector drawables for the category icons.

abdo-essam committed:
refactor: Reduce size of category item title

Updated the typography style for the category item title from
`labelMedium` to `labelSmall` to better fit within the item's
layout and improve visual balance.

abdo-essam committed:
fix: Improve category item check icon color

Updates the color of the check icon in the CategoryItem
component to use the onPrimary color from the TudeeTheme with
an alpha of 0.87. This improves visibility and consistency
with the application's theme.

abdo-essam commented on app/src/main/java/com/london/tudee/presentation/components/CategoryItem.kt:
```diff
@@ -1 +1,140 @@
+                        painter = painterResource(id = R.drawable.ic_check),
+                        contentDescription = null,
+                        modifier = Modifier.size(12.dp),
+                        tint = Color.White
```
Done @Abdulrahman1890

abdo-essam committed:
refactor: Increase CategoryItem icon and badge size

Increase the size of the icon and badge in the
CategoryItem composable for better visibility and
improved user experience. Also slightly adjusted offset and width.

BassantM commented on app/src/main/java/com/london/tudee/presentation/components/CategoryItem.kt:
```diff
@@ -1 +1,141 @@
+            modifier = Modifier
+                .size(78.dp)
+                .background(
+                    color = TudeeTheme.colors.surface,
```
kindly color will be TudeeTheme.colors.surfaceHigh, this will change offset

abdo-essam committed:
feat: Change CategoryItem background color

Update background color of CategoryItem to surfaceHigh for
better visual hierarchy.

abdo-essam committed:
feat: Implement new design system components and assets

This commit introduces new design system components including buttons,
text fields, sliders, and snack bars, along with supporting assets
like icons, images, and Lottie animations. It also removes old
task related files, and modifies color definitions, gradle files
and the commit message hook.

karimdeabes commented on app/src/main/java/com/london/tudee/presentation/components/CategoryItem.kt:
```diff
@@ -1 +1,141 @@
+                        Text(
+                            text = count.toString(),
+                            style = TudeeTheme.typography.labelSmall,
+                            color = TudeeTheme.colors.hint
```
also here

abdo-essam committed:
feat: Implement category item UI components

This commit introduces the `DateItem` and `SwitchButton`
components, along with related resources and color scheme
adjustments, to enhance the category item UI.

- Added `DateItem` composable for displaying dates.
- Added `SwitchButton` composable for toggling settings.
- Added `ellipse_stars.xml` for SwitchButton design
- Modified `LightThemeColors.kt` and `DarkThemeColors.kt` to
  adjust the color scheme for better visual appeal.
- Modified `strings.xml` to remove the redundant app name.
